File: 송현진/백준_JAVA/4358_생태학.java
import java.util.*;
import java.util.Map.Entry;
import java.io.*;

public class Solution {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int totalCnt = 0;
        HashMap<String, Integer> hm = new HashMap<>();

        while (true) {
            String st = br.readLine();
            //eof처리 
            if (st == null || st.length() == 0) {
                break;
            }

            totalCnt++;
            //값없을 시 0으로 디폴트 처리 
            hm.put(st, hm.getOrDefault(st, 0) + 1);
        }

        //priorityQueue 사용해도 될것 같은데 이게 간단해서 사용
        Object[] keys = hm.keySet().toArray();
        Arrays.sort(keys);

        StringBuilder sb = new StringBuilder();

        for (Object key : keys) {
            int val = hm.get(key);
            float f = (float) ((val * 100.0) / totalCnt);
            //소수점 4자리수로 끊기 
            sb.append(key + " " + String.format("%.4f", f) + "\n");
        }
        
        System.out.println(sb);
    }
}

File: 송현진/백준_KOTLIN/4358_생태학.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*
import kotlin.collections.HashMap

fun main() {
    var br = BufferedReader(InputStreamReader(System.`in`))

    var totalCnt = 0
    //자동 정렬 + MAP기능 포함
    var hm = TreeMap<String,Int>()

    while(true){
        var st = br.readLine() ?: break

        totalCnt++
        hm.put(st, hm.getOrDefault(st, 0)+1)
    }


    var sb = StringBuilder()
    hm.forEach{
        var value = hm.getValue(it.key)
        var percent = value * 100.0 / totalCnt
        sb.append("${it.key} ${String.format("%.4f", percent)} \n")
    }

    println(sb)
}